basicrta/tests/test_combine_contacts.py

```diff
@@ -0,0 +1,276 @@
+"""
+Tests for combining contact timeseries from multiple repeat runs.
+"""
+
+import os
+import pytest
+import numpy as np
+import pickle
+
+from basicrta.contacts import CombineContacts
+
+
+class MockAtomGroup:
+    """Mock atom group that can be pickled."""
+    def __init__(self, resids):
+        self.residues = MockResidues(resids)
+        
+
+class MockResidues:
+    """Mock residues that can be pickled."""
+    def __init__(self, resids):
+        self.resids = np.array(resids)
+
+
+
+
+
+@pytest.fixture
+def create_mock_contacts():
+    """Factory fixture for creating mock contact files."""
+    def _create_mock_contacts(tmp_path, filename, n_contacts=100, cutoff=7.0, 
+                           ts=0.1, traj_name="test.xtc", top_name="test.pdb"):
+        """Create a mock contact file for testing."""
+        # Create mock atom groups that can be pickled
+        mock_ag1 = MockAtomGroup([1, 2, 3, 4, 5])
+        mock_ag2 = MockAtomGroup([100, 101, 102])
+        
+        # Create metadata
+        metadata = {
+            'top': top_name,
+            'traj': traj_name,
+            'ag1': mock_ag1,
+            'ag2': mock_ag2, 
+            'ts': ts,
+            'cutoff': cutoff
+        }
+        
+        # Create dtype with metadata
+        dtype = np.dtype(np.float64, metadata=metadata)
+        
+        # Create contact data (4 columns for processed contacts)
+        # [protein_resid, lipid_resid, start_time, residence_time]
+        np.random.seed(42)  # For reproducible tests
+        contacts = np.zeros((n_contacts, 4), dtype=dtype)
+        contacts[:, 0] = np.random.choice([1, 2, 3, 4, 5], n_contacts)  # protein resids
+        contacts[:, 1] = np.random.choice([100, 101, 102], n_contacts)  # lipid resids  
+        contacts[:, 2] = np.random.uniform(0, 100, n_contacts)  # start times
+        contacts[:, 3] = np.random.exponential(1.0, n_contacts)  # residence times
```

Comment:
Sample from a 2-term hyperexponential with probability distribution 
```
p(t) ~ 0.7 * exp(-10.0 * t) + 0.3 * exp(-0.5 * t)
```
instead of a simple exponential with parameter 1.0.